airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
+        Lists long-running operations in Google Storage Transfer
+        Service that match the specified filter.
+
+        :param filter: A request filter, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/list#body.QUERY_PARAMETERS.filter
+        :type filter: dict
+        :return: List of Transfer Jobs
+        :rtype: list[dict]
+        """
+        conn = self.get_conn()
+        filter = self._append_project_id(filter, 'filter project_id parameter')
+        request = conn.transferJobs().list(
+            filter=json.dumps(filter)
+        )
+        jobs = []
+
+        while request is not None:
+            response = request.execute(num_retries=NUM_RETRIES)
+            jobs.extend(response['transferJobs'])
+
+            request = conn.transferJobs().list_next(
+                previous_request=request,
+                previous_response=response)
+
+        return jobs
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def update_transfer_job(self, job_name, body):
+        """
+        Updates a transfer job that runs periodically.
+
+        :param job_name: Name of the job to be updated
+        :type job_name: str
+        :param body: A request body, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/patch#request-body
+        :type body: dict
+        :return: If successful, TransferJob.
+        :rtype: dict
+        """
+        body = self._append_project_id(body, 'body project_id parameter')
+        return self.get_conn()\
+            .transferJobs()\
+            .patch(
+                jobName=job_name,
+                body=body)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    @GoogleCloudBaseHook.convert_http_exception
+    def delete_transfer_job(self, job_name, project_id):
+        """
+        Deletes a transfer job. This is a soft delete. After a transfer job is
+        deleted, the job and all the transfer executions are subject to garbage
+        collection. Transfer jobs become eligible for garbage collection
+        30 days after soft delete.
+
+        :param job_name: Name of the job to be deleted
+        :type job_name: str
+        :param project_id: The ID of the GCP project.
+        :type project_id: str
+        :rtype: None
+        """
+
+        return self.get_conn().transferJobs().patch(
+            jobName=job_name,
+            body={
+                'project_id': project_id,
+                'transfer_job': {
+                    'status': GcpTransferJobsStatus.DELETED
+                },
+                'update_transfer_job_field_mask': 'status'
+            },
+        ).execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def cancel_transfer_operation(self, operation_name):
+        """
+        Cancels an transfer operation in Google Storage Transfer Service.
```

Comment:
```suggestion
        Cancels a transfer operation in Google Storage Transfer Service.
```

Applicable in all docstrings below as well.

Reply:
Done